rabbitmq/src/main/java/io/micronaut/rabbitmq/connect/RabbitConnectionFactory.java

```diff
@@ -55,21 +57,32 @@ public class RabbitConnectionFactory {
     Connection connection(RabbitConnectionFactoryConfig connectionFactory,
                           BeanContext beanContext) {
         try {
-            ExecutorService executorService = beanContext.getBean(ExecutorService.class, Qualifiers.byName(connectionFactory.getConsumerExecutor()));
-            Optional<List<Address>> addresses = connectionFactory.getAddresses();
-            Connection connection;
-            if (addresses.isPresent()) {
-                connection = connectionFactory.newConnection(executorService, addresses.get());
-            } else {
-                connection = connectionFactory.newConnection(executorService);
-            }
+            Connection connection = newConnection(connectionFactory, beanContext);
             activeConnections.add(new ActiveConnection(connection, connectionFactory));
             return connection;
         } catch (IOException | TimeoutException e) {
             throw new BeanInstantiationException("Error creating connection to RabbitMQ", e);
         }
     }
 
+    private Connection newConnection(RabbitConnectionFactoryConfig factory, BeanContext context) throws IOException, TimeoutException {
+        ExecutorService executorService = context.getBean(ExecutorService.class, Qualifiers.byName(factory.getConsumerExecutor()));
+        Optional<List<Address>> addresses = factory.getAddresses();
+        try {
+            if (addresses.isPresent()) {
+                return factory.newConnection(executorService, addresses.get());
+            }
+            return factory.newConnection(executorService);
+        } catch (ConnectException e) {
+            // Check if automatic recovery is enabled
+            if (factory.isAutomaticRecoveryEnabled()) {
+                // Create a "temporarily down" connection that may be up eventually
+                return context.getBean(TemporarilyDownConnectionManager.class).newConnection(executorService);
```

Comment:
Should we not use Constructor injection for `TemporarilyDownConnectionManager`?

Reply:
II don't think we should dynamically fetch things from the context in instances where we can clearly declare injection points.  Please, declare it as a method parameter.